aries_cloudagent/protocols/coordinate_mediation/v1_0/handlers/keylist_request_handler.py

```diff
@@ -0,0 +1,49 @@
+"""Handler for incoming route-list-request messages."""
+
+from .....storage.error import StorageNotFoundError
+from ....problem_report.v1_0.message import ProblemReport
+
+from .....messaging.base_handler import (
+    BaseHandler,
+    BaseResponder,
+    HandlerException,
+    RequestContext,
+)
+
+from ..manager import MediationManager as Manager
+from ..messages.keylist_query import KeylistQuery as request
+from ..models.mediation_record import MediationRecord as _record
+
+
+class KeylistRequestHandler(BaseHandler):
+    """Handler for incoming mediation-keylist-request messages."""
+
+    async def handle(self, context: RequestContext, responder: BaseResponder):
+        """Message handler implementation."""
+        self._logger.debug(
+            "%s called with context %s", self.__class__.__name__, context
+        )
+        assert isinstance(context.message, request)
+
+        if not context.connection_ready:
+            raise HandlerException("Invalid keylist request: no active connection")
+        try:
+            record = await _record.retrieve_by_connection_id(
+                context, context.connection_record.connection_id
+            )
+        except StorageNotFoundError:
+            await self.reject(responder)
+            return
+
+        if context.settings.get("open_mediation", False):
```

Comment:
I think we should be allowing keylist queries whether open mediation is enabled or not. The open mediation flag just tells the agent to automatically grant mediation requests.

Reply:
This check should probably be a check of the mediation state i.e. whether it has been granted. If it hasn't, there shouldn't be any keys to report anyways.

Reply:
so no check needed? I will remove the if statement.

Reply:
The state check is still necessary, just not the open mediation flag check

Reply:
Needed! I will add the check.